src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -422,9 +443,9 @@ private void AddLexedTokenSlot()
             }
             else
             {
-                var tmp = new ArrayElement<SyntaxToken>[_lexedTokens.Length * 2];
-                Array.Copy(_lexedTokens, tmp, _lexedTokens.Length);
-                _lexedTokens = tmp;
+                var old = _lexedTokens;
+                Array.Resize(ref _lexedTokens, _lexedTokens.Length * 2);
+                s_lexedTokensPool.ForgetTrackedObject(old, replacement: _lexedTokens);
```

Comment:
We want to discard if too large and otherwise clear/return the array now, right?

Reply:
Yeah, made a pass through the code again and did some cleanup including this